Rolique/Model/Helpers/ImageManager.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  ImageManager.swift
+//  Rolique
+//
+//  Created by Maksym Ivanyk on 9/2/19.
+//  Copyright © 2019 Rolique. All rights reserved.
+//
+
+import UIKit
+import Foundation
+
+final class ImageManager {
+  private lazy var observer: NSObjectProtocol = {
+    return NotificationCenter.default.addObserver(forName: UIApplication.didReceiveMemoryWarningNotification, object: nil, queue: nil) { [weak self] notification in
+      self?.clearImagesFolder()
+    }
+  }()
+  
+  static let shared = ImageManager()
+  
+  private init() {}
+  
+  deinit {
+    NotificationCenter.default.removeObserver(self.observer)
+  }
+  
+  private let fileManager = FileManager.default
+  private func documentDirectoryURL() throws -> URL {
+    return try fileManager.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false)
+  }
+  private func imagesDirectoryURL() throws -> URL {
```

Comment:
Do you really need to pull this `throws` all the way? Maybe just unwrap it or return optional? Anyway you don't handle `catch` below